qcodes/dataset/measurements.py

```diff
@@ -720,6 +731,27 @@ def _paramspecbase_from_strings(
 
         return tuple(depends_on), tuple(inf_from)
 
+    def register_parent(
+            self: T, parent: DataSet, link_type: str,
+            description: str = "") -> T:
+        """
+        Register a parent for the outcome of this measurement
+
+        Args:
+            parent: the parent dataset
+            link_type: a name for the type of parent-child link
+            description: a free-text description of the relationship
+        """
+        # we save the information in a way that is very compatible with the
+        # Link object we will eventually make out of this information. We
+        # cannot create a Link object just yet, because the DataSet of this
+        # Measurement has not been given a GUID yet
+        parent_dict = {'tail': parent.guid, 'edge_type': link_type,
```

Comment:
shall this then also become a dataclass of its own? say `DataSetReference` or `ParentReference` or else

Reply:
I'd say that's overkill for now. We probably want to change from `dataclass` to some protobuf-based solution once we start evolving the `Link`s. Let's wait until we know more before we implement too much.

Reply:
agree